packages/app-shell-shared/src/hooks/useFilteredConfig.ts

```diff
@@ -0,0 +1,445 @@
+import { useEffect, useState } from "react";
+
+import type { ConditionMetadata, UseConditionResult } from "../types/condition";
+import type {
+  HvAppShellConfig,
+  HvAppShellHeaderAction,
+  HvAppShellMenuConfig,
+  HvAppShellProvidersConfig,
+  HvAppShellViewsConfig,
+} from "../types/Config";
+import { processElementConditions } from "../utils/conditionUtils";
+
+/**
+ * Apply view conditions to menus (inheritance logic)
+ * Merges both explicit menu conditions and inherited view conditions using AND logic
+ */
+const applyViewConditionsToMenus = (
+  menus: HvAppShellMenuConfig[],
+  routeConditionsMap: Map<string, HvAppShellViewsConfig["conditions"]>,
+): HvAppShellMenuConfig[] => {
+  return menus.map((menu) => {
+    let hasChanges = false;
+    let updatedMenu: HvAppShellMenuConfig = menu;
+
+    // If menu has a target, inherit conditions from matching view
+    if (menu.target) {
+      const normalizedTarget = menu.target.replace(/^\//, "");
+      const viewConditions = routeConditionsMap.get(normalizedTarget);
+
+      if (viewConditions) {
+        // Merge explicit menu conditions with inherited view conditions
+        // Both must be satisfied (AND logic)
+        const mergedConditions = [
+          ...(menu.conditions || []),
+          ...viewConditions,
+        ];
+
+        if (mergedConditions.length > 0) {
+          updatedMenu = { ...menu, conditions: mergedConditions };
+          hasChanges = true;
+        }
+      }
+    }
+
+    // Recursively apply to submenus
+    if (menu.submenus) {
+      const updatedSubmenus = applyViewConditionsToMenus(
+        menu.submenus,
+        routeConditionsMap,
+      );
+
+      // Only create new object if submenus changed
+      if (updatedSubmenus !== menu.submenus) {
+        updatedMenu = hasChanges
+          ? { ...updatedMenu, submenus: updatedSubmenus }
+          : { ...menu, submenus: updatedSubmenus };
+        hasChanges = true;
+      }
+    }
+
+    return hasChanges ? updatedMenu : menu;
+  });
+};
+
+/**
+ * Create a map of route -> conditions from views for menu inheritance
+ */
+const createRouteConditionsMap = (
+  views: HvAppShellViewsConfig[],
+): Map<string, HvAppShellViewsConfig["conditions"]> => {
+  const map = new Map<string, HvAppShellViewsConfig["conditions"]>();
+
+  const processView = (view: HvAppShellViewsConfig) => {
+    if (view.route && view.conditions) {
+      const normalizedRoute = view.route.replace(/^\//, "");
+      map.set(normalizedRoute, view.conditions);
+    }
+    if (view.views) {
+      for (const nestedView of view.views) {
+        processView(nestedView);
+      }
+    }
+  };
+
+  for (const view of views) {
+    processView(view);
+  }
+  return map;
+};
+
+export const extractConditionsMetadata = (
```

Comment:
Given the contents of the function, and its nested functions, wouldn't this better be called something like `processConditions`?